modules/salt/pages/salt-big-scale.adoc

```diff
@@ -0,0 +1,401 @@
+
+Then, consult the "Dependency graph" below:
+
+* Tuning inputs are represented by graph nodes in a rectangle on top
+* each node is connected with arrows to parameters that might need tuning. Consult the relevant parameter sections below for more information about recommended values
+* tuning one parameter might require tuning other parameters, or changing hardware, or the infrastructure. Follow the arrows going out of each node that represent a parameter that has changed and repeat the process until no more nodes need to be visited
+
+===== Re-tuning an existing installation
+
+Re-tuning should be attempted:
+
+ - whenever "Tuning inputs" (see First installation) change significantly,
+ - if some special conditions arise that require a certain parameter to be changed (eg. specific warnings appear in logs, as detailed below) or
+ - if performance is not satisfactory.
+
+In that case, re-visit the "Dependency graph" following the same procedure described in First installation, starting from the nodes where significant change has happened.
+
+===== Dependency graph
+
+image::parameter_graph.svg[Tuning dependency graph]
+
+Key:
+
+* 3d boxes represent hardware design variables or constraints
+* ellipses represent software/system design variables or constraints
+* rectangles represent configurable parameters, color-coded by configuration file
+** red: Apache `httpd` configuration files
+** blue: Salt configuration files
+** brown: Tomcat configuration files
+** grey: PostgreSQL configuration files
+** violet: `/etc/rhn/rhn.conf`
+* dashed arrows: a variable or constraint might require a change to another parameter
+* solid arrows: changing a configuration parameter requires checking another one, as failure to adapt it might create issues
+
+
+==== Tuning inputs
+
+===== Network Bandwidth
+
+A measure of the typically available egress bandwith from the {productname} Server host (to the clients or {productname} Proxy hosts). This should take into account network hardware and topology as well as possible capacity limits on switches, routers, and other network equipment between the Server and clients.
+
+===== Channel count
+
+The number of expected channels to manage, including any vendor-provided, third-party and clones/staged channels.
+
+===== Client count
+
+The total number of expected clients.
+
+===== OS mix
+
+The number of distinct OS versions that managed clients have installed, differentiated by family (SUSE Linux Enterprise-based, openSUSE-based, Red Hat-based or Ubuntu-based). Storage and computing requirements are different in each case.
+
+===== User count
+
+The expected maximum amount of concurrent users interacting with the Web UI plus the number of programs simultaneously using the XMLRPC API (including `spacecmd`, `spacewalk-clone-by-date`, etc.).
+
+==== Parameters
+
+===== `MaxClients`
```

Comment:
Nitpick: Maybe better to mention it explicitly that `MaxClients` has been renamed to `MaxRequestWorkers`